File: Haskell/bogoSort.hs
import System.Random (randomRIO, newStdGen)
import Data.List (permutations, sort)
import Control.Monad (replicateM)

isSorted :: Ord a => [a] -> Bool
isSorted [] = True
isSorted [_] = True
isSorted (x:y:xs) = x <= y && isSorted (y:xs)

shuffle :: [a] -> IO [a]
shuffle xs = do
    gen <- newStdGen
    let n = length xs
    indices <- replicateM n (randomRIO (0, n - 1))
    return $ map (\i -> xs !! i) (take n (permutations indices !! 0)) -- Simplified for demonstration

bogosort :: Ord a => [a] -> IO [a]
bogosort xs = do
    shuffledList <- shuffle xs
    if isSorted shuffledList
        then return shuffledList
        else bogosort xs
        

File: Python/Algorithm/bogo_sort.py
import random

def shuffle(array):
    if len(array) < 0:
        return None
    
    for i in range(len(array)-1, 0, -1):
        j = random.randint(0, i)

        array[i], array[j] = array[j], array[i]

    return array

def is_sorted(array):
    for i in range(0, len(array)-1):
        if array[i] > array[i+1]:
            return False
    return True

def bogo_sort(array):
    new_array = array
    while not is_sorted(new_array):
        new_array = shuffle(new_array)
    return new_array


print(bogo_sort([5, 4, 3, 2, 1]))